ext/pdo_oci/tests/pdo_oci_stmt_getcolumnmeta.phpt

```diff
@@ -0,0 +1,281 @@
+--TEST--
+PDO_OCI: PDOStatement->getColumnMeta
+--SKIPIF--
+<?php # vim:ft=php
+if (!extension_loaded('pdo') || !extension_loaded('pdo_oci')) die('skip not loaded');
+require(dirname(__FILE__) . '/../../pdo/tests/pdo_test.inc');
+PDOTest::skip();
+?>
+--FILE--
+<?php
+
+echo "Preparations before the test\n";
+
+require(dirname(__FILE__) . '/../../pdo/tests/pdo_test.inc');
+try {
+	$db = PDOTest::factory();
+	$db->exec(<<<SQL
+BEGIN
+   EXECUTE IMMEDIATE 'DROP TABLE test';
+EXCEPTION
+   WHEN OTHERS THEN
+      IF SQLCODE != -942 THEN
+	 RAISE;
+      END IF;
+END;
+SQL
+);
+	$db->exec("CREATE TABLE test(id INT)");
+
+	$db->beginTransaction();
+
+	$stmt = $db->prepare('SELECT id FROM test ORDER BY id ASC');
+
+	echo "Test 1. calling function with invalid parameters\n";
+
+	// execute() has not been called yet
+	// NOTE: no warning
+	if (false !== ($tmp = $stmt->getColumnMeta(0)))
+		printf("[002] Expecting false got %s\n", var_export($tmp, true));
+
+	$stmt->execute();
+	// Warning: PDOStatement::getColumnMeta() expects exactly 1 parameter, 0 given in
+	if (false !== ($tmp = @$stmt->getColumnMeta()))
+		printf("[003] Expecting false got %s\n", var_export($tmp, true));
+
+	// invalid offset
+	if (false !== ($tmp = @$stmt->getColumnMeta(-1)))
+		printf("[004] Expecting false got %s\n", var_export($tmp, true));
+
+	// Warning: PDOStatement::getColumnMeta() expects parameter 1 to be int, array given in
+	if (false !== ($tmp = @$stmt->getColumnMeta(array())))
+		printf("[005] Expecting false got %s\n", var_export($tmp, true));
+
+	// Warning: PDOStatement::getColumnMeta() expects exactly 1 parameter, 2 given in
+	if (false !== ($tmp = @$stmt->getColumnMeta(1, 1)))
+		printf("[006] Expecting false got %s\n", var_export($tmp, true));
+
+	// invalid offset
+	if (false !== ($tmp = $stmt->getColumnMeta(1)))
+		printf("[007] Expecting false because of invalid offset got %s\n", var_export($tmp, true));
+
+	echo "Test 2. testing return values\n";
+	echo "Test 2.1 testing array returned\n";
+
+	$stmt = $db->prepare('SELECT id FROM test ORDER BY id ASC');
+	$stmt->execute();
+	$native = $stmt->getColumnMeta(0);
+	if (count($native) == 0) {
+		printf("[008] Meta data seems wrong, %s / %s\n",
+			var_export($native, true), var_export($emulated, true));
+	}
+
+	function test_meta(&$db, $offset, $sql_type, $value, $native_type, $pdo_type) {
+
+		$db->exec(<<<SQL
+BEGIN
+   EXECUTE IMMEDIATE 'DROP TABLE test';
+EXCEPTION
+   WHEN OTHERS THEN
+      IF SQLCODE != -942 THEN
+         RAISE;
+      END IF;
+END;
+SQL
+);
+
+		$sql = sprintf('CREATE TABLE test(id INT, label %s)', $sql_type);
+		$stmt = $db->prepare($sql);
+		$stmt->execute();
+
+		if (!$db->exec(sprintf("INSERT INTO test(id, label) VALUES (1, '%s')", $value))) {
```

Comment:
We strongly advise developers not to use string concatenation for SQL since it reduces statement caching (i.e impacts performance and scalability) and is open to SQL Injection security attacks.  Instead, binding is preferred.  In the context of this test, the concatenation is OK to do, but not a habit to continue with!